src/util/statistics/counter/perf_event.rs

```diff
@@ -9,9 +9,10 @@ pub struct PerfEventDiffable {
 }
 
 impl PerfEventDiffable {
-    pub fn new(name: &str) -> Self {
+    pub fn new(name: &str, exclude_kernel: bool) -> Self {
         let mut pe = PerfEvent::new(name, true)
             .unwrap_or_else(|_| panic!("Failed to create perf event {}", name));
+        pe.set_exclude_kernel(exclude_kernel as u64);
```

Comment:
Since we set `PFM_PLM3` only, does that mean all the kernel events are already being excluded, even when `set_exclude_kernel` is not called?

Reply:
The function [`pfm_get_perf_event_encoding`](https://man7.org/linux/man-pages/man3/pfm_get_perf_event_encoding.3.html) is only creating the `perf_event_attr`. Since by default, only `PFM_PLM3` is passed, it will only be enabled for user-space events (so both `exclude_{kernel,hv}` are set to `1`). If we then set the `exclude_kernel` to `0` later ourselves (like this PR is doing), then it should be including kernel events.

Reply:
Side-note, it seems like `pfm_get_perf_event_encoding` is deprecated and it's recommended to use [`pfm_get_os_event_encoding`](https://man7.org/linux/man-pages/man3/pfm_get_os_event_encoding.3.html) instead.